### PR TITLE
[FEATURE] Add custom placeholder processing in YAML

### PR DESCRIPTION
**Pull request by susannemoog**
See Core Feature 90267
See #845

### CODE REVIEW

File: Documentation/ApiOverview/Yaml/Index.rst [hunk 1]
```diff
@@ -56,3 +56,64 @@ Examples:
...
+
+   $GLOBALS['TYPO3_CONF_VARS']['SYS']['yamlLoader']['placeholderProcessors'][\Vendor\MyExtension\PlaceholderProcessor\CustomPlaceholderProcessor::class] = [
+      'before' => [
+         \TYPO3\CMS\Core\Configuration\Processor\Placeholder\ValueFromReferenceArrayProcessor::class
+      ],
+      'after' => [
+         \TYPO3\CMS\Core\Configuration\Processor\Placeholder\EnvVariableProcessor::class
+      ],
+      'disabled' => false
+   ];
```
Comment (mbrodala) [inline Documentation/ApiOverview/Yaml/Index.rst:83]: Git loves trailing commas for each array item. ;-)

File: Documentation/ApiOverview/Yaml/Index.rst [hunk 2]
```diff
@@ -56,3 +56,64 @@ Examples:
...
+   {
+      public function canProcess(string $placeholder, array $referenceArray): bool
+      {
+         return strpos($placeholder, '%example(') !== false;
+      }
+
+      public function process(string $value, array $referenceArray)
+      {
+         // do some processing
+         $result = $this->getValue($value);
```
Comment (mbrodala) [inline Documentation/ApiOverview/Yaml/Index.rst:100]: We should add at least a simulated method for this.

File: Documentation/ApiOverview/Yaml/Index.rst [hunk 3]
```diff
@@ -56,3 +56,64 @@ Examples:
...
+
+         // Throw this exception if the placeholder can't be substituted
+         if (!$envVar) {
```
Comment (mbrodala) [inline Documentation/ApiOverview/Yaml/Index.rst:103]: Where does this come from suddenly? ;-)

**Review by mbrodala — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/Yaml/Index.rst (hunk 2 above)
Comment (susannemoog) [reply Documentation/ApiOverview/Yaml/Index.rst:100]: tried to add one, it doesn't make much sense but I couldn't think of a better one.

File: Documentation/ApiOverview/Yaml/Index.rst (hunk 3 above)
Comment (susannemoog) [reply Documentation/ApiOverview/Yaml/Index.rst]: My clipboard, definitely my clipboard ;)

**Review by mbrodala — APPROVED** [review]

**Pull request merged**